### PR TITLE
Mention bind-mounting plugins dir in Docker docs

### LINKED ISSUE #69533: Verify Elasticsearch Docker container writes on plugin installation
**Elasticsearch version**: `8.0.0-SNAPSHOT` 

**OS version**: Docker container

When running Elasticsearch Docker container with **an arbitrary UID/GID** it's impossible to install a plugin:
```
$ docker run --user 31337:1337 -i -t docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT bash
bash-4.4$ ./bin/elasticsearch-plugin install --batch https://snapshots.elastic.co/8.0.0-000145a6/downloads/elasticsearch-plugins/repository-s3/repository-s3-8.0.0-SNAPSHOT.zip
-> Installing https://snapshots.elastic.co/8.0.0-000145a6/downloads/elasticsearch-plugins/repository-s3/repository-s3-8.0.0-SNAPSHOT.zip
-> Downloading https://snapshots.elastic.co/8.0.0-000145a6/downloads/elasticsearch-plugins/repository-s3/repository-s3-8.0.0-SNAPSHOT.zip
-> Failed installing https://snapshots.elastic.co/8.0.0-000145a6/downloads/elasticsearch-plugins/repository-s3/repository-s3-8.0.0-SNAPSHOT.zip
-> Rolling back https://snapshots.elastic.co/8.0.0-000145a6/downloads/elasticsearch-plugins/repository-s3/repository-s3-8.0.0-SNAPSHOT.zip
-> Rolled back https://snapshots.elastic.co/8.0.0-000145a6/downloads/elasticsearch-plugins/repository-s3/repository-s3-8.0.0-SNAPSHOT.zip
Exception in thread "main" java.nio.file.AccessDeniedException: /usr/share/elasticsearch/plugins/.installing-668324506119825940
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:90)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:388)
	at java.base/java.nio.file.Files.createDirectory(Files.java:694)
	at java.base/java.nio.file.TempFileHelper.create(TempFileHelper.java:135)
	at java.base/java.nio.file.TempFileHelper.createTempDirectory(TempFileHelper.java:172)
	at java.base/java.nio.file.Files.createTempDirectory(Files.java:970)
	at org.elasticsearch.plugins.InstallPluginCommand.stagingDirectory(InstallPluginCommand.java:738)
	at org.elasticsearch.plugins.InstallPluginCommand.unzip(InstallPluginCommand.java:685)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:227)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:199)
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:75)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:116)
	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:80)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:116)
	at org.elasticsearch.cli.Command.main(Command.java:79)
	at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:36)
``` 

According to [docs](https://github.com/elastic/elasticsearch/blob/master/docs/reference/setup/install/docker.asciidoc), there is a handy workaround for this - since `/usr/share/elasticsearch` is group-writable user can leverage the `--group-add 0` option. 

That is described as a desired way to go **unless** you bind-mount config/data/logs directory. The `./bin/elasticsearch-plugin` tool however, writes outside of those locations.

### PR DESCRIPTION
**Pull request by pugnascotia**
Closes #69533.

The Docker docs mention bind-mounting the `config`, `data` and `logs` directories when using an arbitrary UID / GID, but they fail to mention that the `plugins` dir must also be mounted in order to install plugins.

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]
LGTM. Thanks, Rory.

File: docs/reference/setup/install/docker.asciidoc [hunk 1]
```diff
@@ -425,11 +425,17 @@ chmod g+rwx esdatadir
...
 --------------------------------------------
 
-You can also run an {es} container using both a custom UID and GID. Unless you
-bind-mount each of the `config`, `data` and `logs` directories, you must pass
-the command line option `--group-add 0` to `docker run`. This ensures that the user
-under which {es} is running is also a member of the `root` (GID 0) group inside the
-container.
+You can also run an {es} container using both a custom UID and GID. You
+must ensure that file permissions will not prevent {es} from executing. You
+have two options.
```
Comment (jrodewig) [inline docs/reference/setup/install/docker.asciidoc:430]: I'd add a semicolon to the end and emphasize you only need to do one of these.
```suggestion
must ensure that file permissions will not prevent {es} from executing. You
can use one of two options:
```

File: docs/reference/setup/install/docker.asciidoc [hunk 2]
```diff
@@ -425,11 +425,17 @@ chmod g+rwx esdatadir
...
+
+* Pass the `--group-add 0` command line options to `docker run`. This
```
Comment (jrodewig) [inline docs/reference/setup/install/docker.asciidoc:432]: I think this is a singular option.
```suggestion
* Pass the `--group-add 0` command line option to `docker run`. This
```

File: docs/reference/setup/install/docker.asciidoc [hunk 3]
```diff
@@ -425,11 +425,17 @@ chmod g+rwx esdatadir
...
+  ensures that the user under which {es} is running is also a member of the
+  `root` (GID 0) group inside the container.
+* Alternatively, you must bind-mound the `config`, `data` and `logs`
```
Comment (jrodewig) [inline docs/reference/setup/install/docker.asciidoc:435]: We can just use the imperative here.
```suggestion
* Bind-mound the `config`, `data` and `logs`
```

File: docs/reference/setup/install/docker.asciidoc [hunk 4]
```diff
@@ -425,11 +425,17 @@ chmod g+rwx esdatadir
...
+  directories. You must also bind-mount the `plugins` directory if you
+  intend to install plugins and prefer not to <<_c_customized_image, create a
+  custom Docker image>>.
```
Comment (jrodewig) [inline docs/reference/setup/install/docker.asciidoc:438]: I'd flip the order here so the user's intent comes first.
```suggestion
  directories. If you intend to install plugins and prefer not to
  <<_c_customized_image, create a custom Docker image>>, you must also
  bind-mount the `plugins` directory.
```

Comment (jrodewig) [inline docs/reference/setup/install/docker.asciidoc:438]: Should we move the bind-mounting bullet first? I imagine most admins are using Docker compose in production so I'm not sure if we want to highlight the `docker run` option first.

**Review by jrodewig — APPROVED** [review]
LGTM. I left some suggestions and comments, but none of those are blocking.

File: docs/reference/setup/install/docker.asciidoc (hunk 4 above)
Comment (pugnascotia) [reply docs/reference/setup/install/docker.asciidoc]: I'm honestly not sure which option an admin would prefer, but I do suspect that people following the guide to get something up-and-running would be better off using the bind-mounts, so I've swapped the options.

**Comment by pugnascotia** [comment]
Thanks @jrodewig!

File: docs/reference/setup/install/docker.asciidoc (hunk 4 above)
Comment (mark-vieira) [reply docs/reference/setup/install/docker.asciidoc]: >  I imagine most admins are using Docker compose in production so I'm not sure if we want to highlight the docker run option first.

FWIW, I don't think folks using our Docker images in production are using either of these options. They are likely using Kubernetes or some other orchestrator or config as code mechanism to deploy. That's just my intuition though.

**Pull request merged**